### DISCUSSION TITLE
Echidna 2.2.0

### DISCUSSION THREAD

**Discussion by arcz** [crytic/echidna#1053] [Announcements]
Echidna 2.2.0 contains significant improvements to the fuzzing speed and UX:
* **Multicore fuzzing & optimized coverage collection.** Those combined delivered up to 20x fuzzing speed improvement on a real-world internal benchmark. The number of workers can be configured with `workers` config option or `--workers` CLI switch. Echidna runs only one worker by default, but this might change in future releases.
* **Lcov support.** It is output by default and can be controlled with the [coverageFormats config option](https://github.com/crytic/echidna/blob/v2.2.0/tests/solidity/basic/default.yaml#L81). This enables external coverage tools such as [genhtml](https://linux.die.net/man/1/genhtml) or [VSCode plugins](https://marketplace.visualstudio.com/items?itemName=ryanluker.vscode-coverage-gutters).
* **More configuration options.** Added `--timeout <seconds>` CLI switch. RPC URL and block number can now be also [specified in the config file](https://github.com/crytic/echidna/blob/v2.2.0/tests/solidity/basic/default.yaml#L86-L89) for on-chain fuzzing.
* **UI improvements.** Echidna now outputs an event log for any fuzzing breakthrough, such as new coverage. The interactive UI was significantly reworked to accommodate all the changes from this release (see the screenshot below).

<img width="855" alt="Screenshot 2023-05-21 at 20 30 03" src="https://github.com/crytic/echidna/assets/4679721/30439729-9ddb-4b70-9e50-316390e7b4de">

Note, we changed the way reverts are shown in coverage reports. Now, only the line where a revert happened is marked, instead of the whole path.

The full changelog:

## Added

* Multicore fuzzing (#963, #1033, #1026, #1035)
* Lcov format support (#1029)
* Experimental power number generator for uints (#892)

## Changed

* Coverage collection optimization (#1003, #1041)
* On-chain fuzzing improvements (#1017)
* Refactored code (#1020, #1021)
* Updated dependencies (#1022, #1023)
* UI improvements (#1031, #1032, #1034, #1040)
* Readme improvements (#1019)

<hr /><em>This discussion was created from the release <a href='https://github.com/crytic/echidna/releases/tag/v2.2.0'>Echidna 2.2.0</a>.</em>

**Comment by rayeaster** [comment]
@arcz Hello ser, is there a precompiled windows file for this version? Maybe this one https://github.com/crytic/echidna/actions/runs/5039460033 ?

**Reply by arcz** [reply]
I've just added the Windows binary :)
